working_villagers/jobs/dyemixer.lua

```diff
@@ -0,0 +1,1489 @@
+local func = working_villages.require("jobs/util")
+local log = working_villages.require("log")
+
+function mod(x, m)
+	assert(x ~= nil)
+	assert(m ~= nil)
+	local r = x % m
+	--if r < 0 then
+	--	r = r+m
+	--end
+	r = r % m
+	assert(0 <= r)
+	assert(r <  m)
+	return r
+end
+
+local dyemixers = {
+	names = {
+		["mcg_dyemixer:dye_mixer"]={},
+	},
+}
+
+local wools = {
+	dye_groups = {
+	},
+	groups = {
+		["wool"]=99,
+	},
+}
+
+-- TODO write a dependency management system
```

Comment:
This sounds like a great idea, since this function contains a lot of definitions that should not be part of the job code.
Even if the management system is developed later, I think this code should move out of the way, into another file, so this job is more readable.

Reply:
I created a dyemixer_recipes.lua. It should be easy to replace it when the time comes. Makefile-like dependency tracking will probably help with the craftsmen and logistics/supplychain bots. This is gonna take a lot of coffee to write.